File: fibonacci/fibonacci.zig
const std = @import("std");

fn fibonacci(n: u32) u128 {
    var a: u128 = 0;
    var b: u128 = 1;
    var i: u32 = 0;
    
    while (i < n) : (i += 1) {
        const temp = a;
        a = b;
        b += temp;
    }
    
    return a;
}

pub fn main() !void {
    const result = fibonacci(100000);
    const stdout = std.io.getStdOut().writer();
    try stdout.print("{}\n", .{result});
}

File: fibonacci/fibonacci.rs
fn fibonacci(n: u32) -> u64 {
    let mut a = 0u64;
    let mut b = 1u64;
    for _ in 0..n {
        let temp = a;
        a = b;
        b += temp;
    }
    a
}

fn main() {
    let result = fibonacci(100000);
    println!("{}", result);
}